### PR TITLE
Add `CGAL/version_checker.h`

### PR DESCRIPTION
**Pull request by janetournois**
## Summary of Changes

Following discussion #7525 I added the header `version_enforcer.h`

In the general case, it has nothing to check. If the user includes his own fork of `version_enforcer.h` with a given number of CGAL version, then the check happens.

## Release Management

* Affected package(s): Installation

### CODE REVIEW

File: Installation/include/CGAL/version_enforcer.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// you want to use.
+#define CGAL_AUTHORIZED_VERSION CGAL_VERSION_STR
+#define CGAL_AUTHORIZED_VERSION_MAJOR CGAL_VERSION_MAJOR
+#define CGAL_AUTHORIZED_VERSION_MINOR CGAL_VERSION_MINOR
+#define CGAL_AUTHORIZED_VERSION_PATCH CGAL_VERSION_PATCH
+
+// Check that the version of CGAL used is the one expected
+#if (CGAL_VERSION_MAJOR != CGAL_AUTHORIZED_VERSION_MAJOR)
+#pragma message "You are using CGAL version: " CGAL_STR(CGAL_VERSION) "."
+#error This header is meant to be with used with CGAL 5.5 only.
```
Comment (albert-github) [inline Installation/include/CGAL/version_enforcer.h:28]: In my opinion a bit strange that in the error text the value `5.5` is mentioned whilst the test is more flexible.

Comment (janetournois) [reply Installation/include/CGAL/version_enforcer.h:28]: you are right! Thanks

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-20

File: Installation/include/CGAL/version_checker.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#define CGAL_VERSION_CHECKER_ERROR 0
+#define CGAL_VERSION_CHECKER_WARNING 0
+
+#define CGAL_AUTHORIZED_VERSION_STR CGAL_STR(CGAL_AUTHORIZED_VERSION_MAJOR) "." \
+                                    CGAL_STR(CGAL_AUTHORIZED_VERSION_MINOR) "." \
+                                    CGAL_STR(CGAL_AUTHORIZED_VERSION_PATCH)
+
+
+// Check that the version of CGAL used is the one expected
+#if  CGAL_AUTHORIZED_VERSION_MAJOR != CGAL_VERSION_MAJOR \
```
Comment (sloriot) [inline Installation/include/CGAL/version_checker.h:37]: AUTHORIZED -> EXPECTED?

Comment (MaelRL) [reply Installation/include/CGAL/version_checker.h:37]: I'd put something like "compatible" and replace "bad version" / "wrong version" / "authorized version" to "compatible" / "incompatible"

**Comment by lrineau** [comment]
This PR has the milestone CGAL-5.6 but its base branch is `master` (for CGAL-6.0). Let me know what I should do.

**Comment by janetournois** [comment]
If there are more things to be merged to 5.6, it can be merged there. Otherwise I can wait for the next release, no worries

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-33

**Review by sloriot — APPROVED** [review]

**Pull request merged**